native/core/src/execution/planner.rs

```diff
@@ -1208,7 +1208,18 @@ impl PhysicalPlanner {
 
                 // By default, local FS object store registered
                 // if `hdfs` feature enabled then HDFS file object store registered
-                let object_store_url = register_object_store(Arc::clone(&self.session_ctx))?;
+                // Get one file from the list of files
+                let one_file = scan
+                    .file_partitions
+                    .first()
+                    .unwrap()
+                    .partitioned_file
+                    .first()
+                    .unwrap()
+                    .file_path
+                    .clone();
```

Comment:
We can avoid the unwraps here and add error handling:

```suggestion
                let one_file = scan
                    .file_partitions
                    .first()
                    .and_then(|f| f.partitioned_file.first())
                    .map(|f| f.file_path.clone())
                    .ok_or(ExecutionError::GeneralError(
                        "Failed to locate file".to_string(),
                    ))?;
```